### PR TITLE
Switch from Dep to Go Modules

### PR DESCRIPTION
**Pull request by marinalimeira**
This is the first time I do this, so not sure if I missed something!

Followed [these](https://stackoverflow.com/a/55664631) instructions.

### CODE REVIEW

**Review by yorinasub17 — COMMENTED** [review]
Thanks for doing this! Can you also run `go mod tidy` so that it creates a `go.sum` file?

**Comment by yorinasub17** [comment]
Ohhhhhh waittttt. I just realized I had attempted this earlier, and failed to get the tests to pass: https://github.com/gruntwork-io/package-terraform-utilities/pull/31

I think there is a problem with the circleci docker image and go modules.

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,7 +1,14 @@
 defaults: &defaults
   working_directory: /home/circleci/.go_workspace/src/github.com/gruntwork-io/package-terraform-utilities
```
Comment (yorinasub17) [inline .circleci/config.yml:2]: You need to remove this if you are using the machine image.

**Comment by marinalimeira** [comment]
@yorinasub17 I am doing exactly the same thing you did before :D

**Comment by yorinasub17** [comment]
> I am doing exactly the same thing you did before

There's actually one significant difference you are attempting here, which is that you are running it on the machine image as opposed to docker, so I think this has a good chance of succeeding :)

**Comment by marinalimeira** [comment]
I started this because the Terratest version is too old and it was missing some features and it looks like it wasn't as simple as I thought

>  which is that you are running it on the machine image as opposed to docker

Hope it works! 🤞

**Comment by marinalimeira** [comment]
Made a rebase since I had many 1-line commits to test the CircleCI build.

**Comment by marinalimeira** [comment]
@yorinasub17 the build is working 🎉 🎉 🎉 🎉 🎉

**Review by infraredgirl — APPROVED** [review]
LGTM!

**Comment by marinalimeira** [comment]
Thanks for the review Ana!

Merging :)

**Comment by marinalimeira** [comment]
Will release a new version after merge #34

**Pull request merged**